django/db/models/sql/where.py

```diff
@@ -76,6 +77,12 @@ def as_sql(self, compiler, connection):
         else:
             full_needed, empty_needed = 1, len(self.children)
 
+        if self.connector == XOR and not connection.features.supports_logical_xor:
+            # Convert `A XOR B` to `(A OR B) AND NOT (A AND B)`.
+            lhs = self.__class__(self.children, OR)
+            rhs = self.__class__(self.children, AND, negated=True)
```

Comment:
This doesn't work with many conditions, e.g. `a ^ b ^ c` see `XorLookupsTests.test_filter_negated()`.

Reply:
Great catch. 

It looks like 3,4,.. args would need some higher-level logic. I could accomplish that by going through the children in search of `XOR`s, and their children, and so on... and then building up equivalent logic from that list

I will work on that tonight when I have free time after work, it should not be too difficult. Though I'm already stressing about how I'm going to keep it clean.

Reply:
It was easier than I thought. instead of `(A OR B) AND NOT (A AND B)`, the new logic is:
`(A OR B OR ...) AND 1 = (CASE(1 if A else 0) + CASE(1 if B else 0) + ...)`

NOTE: I tried to use `Cast` or `ExpressionWrapper` instead but they did not seem to work